### ISSUE TITLE
Constant indices and @ArrayLen; also refinement to learn @ArrayLen

### ISSUE THREAD

**Issue by mernst** [kelloggm/checker-framework#64]
All 4 of the array indexing operations yield a warning about being possibly too high.
This file is checked in as `checker-framework/checker/tests/upperbound/LengthTransfer2.java`.

```java
class LengthTransfer {
    public static void main(String[] args) {
      if (args.length != 2) {
        System.err.println("Needs 2 arguments, got " + args.length);
      }
      int limit = Integer.parseInt(args[0]);
      int period = Integer.parseInt(args[1]);
    }

    void m(String @ArrayLen(2) [] args) {
      int limit = Integer.parseInt(args[0]);
      int period = Integer.parseInt(args[1]);
    }
}
```

**Comment by kelloggm** [comment]
I agree that the second pair of accesses are spurious warnings. However, I think the first pair are correct, since the System.err.println call doesn't result in termination. If we put a return in that block, and we still issue warnings, then it's wrong, but as is if there are fewer than two arguments the code in main will issue the warning message and then go ahead and crash with an IndexOutOfBounds

**Comment by mernst** [comment]
You are right about the first two warnings.  (I had simplified the test case.)  I have updated the file in the repository.

The last two warnings remain, and are spurious; they are what this issue is really about.

**Comment by kelloggm** [comment]
Closed by cfaf7f5d4ba0d900804aeeb029e38c97ef0580d4
